torch/_export/utils.py

```diff
@@ -1350,3 +1350,52 @@ def default_flatten_fn_spec(obj, spec) -> list[Any]:
 def deregister_module_as_pytree_input_node(cls: type[torch.nn.Module]) -> None:
     _deregister_pytree_node(cls)
     _deregister_pytree_flatten_spec(cls)
+
+
+def _sync_state(src, dst):
+    assert isinstance(
+        src,
+        torch.nn.Module,
+    ), f"Expected {src} to be a nn.Module"
+    assert isinstance(
+        dst,
+        torch.nn.Module,
+    ), f"Expected {dst} to be a nn.Module"
+    # Share state (params, buffers) between modules.
+    # This ensures that state mutations are visible across them.
+    # Since tensor constants are not mutable, copying (without sharing) is OK.
+    # Also, primitive constants are specialized, so copying (without sharing) is OK.
+    dst._parameters = src._parameters
```

Comment:
Instead of relying on private attributes, should we use public APIs like module.register_param etc? Or do they implicitly always copy?

Reply:
Having the same `_parameters` and `_buffers` objects is an easy way to ensure that everything is always in sync when you use the original model. We use this trick in unflatten as well. I'm not sure whether registering will have the same effect, i.e., `foo` and `bar` modules may not track changes to the original module's state?